inst/@sym/cumprod.m

```diff
@@ -0,0 +1,85 @@
+%% Copyright (C) 2020 Tasos Papastylianou
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @defmethod  @@sym cumprod (@var{f})
+%% @defmethodx @@sym cumprod (@var{f}, @var{n})
+%% Cumulative product of symbolic expressions.
+%%
+%% Perform a Cumulative Product over the rows or columns of an
+%% expression. By default, calculate product over rows.  Can specify
+%% row or column product using @var{n}.
+%%
+%% Examples:
+%% @example
+%% @group
+%% t = sym( 't' );
+%% Pi = sym( 'pi' );
+%% T = [ t; t + Pi / 2; t + Pi; t + 3 * Pi / 2; t + 2 * Pi ];
+%% C = cos(T);
+%% cumprod( C )
+%%   @result{} (sym 5×1 matrix)
+%%       ⎡    cos(t)     ⎤
+%%       ⎢               ⎥
+%%       ⎢-sin(t)⋅cos(t) ⎥
+%%       ⎢               ⎥
+%%       ⎢          2    ⎥
+%%       ⎢sin(t)⋅cos (t) ⎥
+%%       ⎢               ⎥
+%%       ⎢   2       2   ⎥
+%%       ⎢sin (t)⋅cos (t)⎥
+%%       ⎢               ⎥
+%%       ⎢   2       3   ⎥
+%%       ⎣sin (t)⋅cos (t)⎦
+%% @end group
+%% @end example
+%%
+%% Specify dimension along which to sum:
+%% @example
+%% @group
+%% X = repmat( sym( 'x' ), 1, 5 );
+%% cumprod( X, 2 )
+%%   @result{} (sym 1×5 matrix)
+%%       ⎡    2   3   4   5⎤
+%%       ⎣x  x   x   x   x ⎦
+%% @end group
+%% @end example
+%%
+%% @seealso{@@sym/prod, @@sym/cumsum}
+%% @end defmethod
+
+
+function Y = cumprod( X, N )
+
+    if nargin < 2
```

Comment:
Please use parens on conditionals, and indentation should be 2 spaces

Reply:
You mean like `if (nargin < 2)`? I was not aware this was the official style ...
Do you have a link to the official style?

Reply:
https://wiki.octave.org/Octave_style_guide